### PR TITLE
feat: Add VerifyExpressions API

### PR DESCRIPTION
**Pull request by manitgupta**
Implements the VerifyExpressions API.

### CODE REVIEW

File: expressions_api/expression_verify.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			return fmt.Errorf("one of expressionId, expression, type or referenceElement.Name is empty. These are mandatory fields = %v", expressionDetail)
+		}
+	}
+	return nil
+}
+
+// We simplify conv to remove any existing expressions that are part of the SpSchema to ensure that the stagingDB creation
+// does not fail due to inconsistent, user configured expressions during a schema conversion session.
+// The minimal conv object needed for stagingDB is one which contains all table and column definitions only.
+func (ev *ExpressionVerificationAccessorImpl) simplifyConv(inputConv *internal.Conv) (*internal.Conv, error) {
```
Comment (bharadwaj-aditya) [inline expressions_api/expression_verify.go:113]: nit:rename to removeExpressions

**Review by bharadwaj-aditya — COMMENTED** [review]
minor comments.

File: expressions_api/expression_verify.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+	spannerclient "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/spanner/client"
+	spanneraccessor "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/spanner"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/common"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/spanner/ddl"
+)
+
+const THREAD_POOL = 500
```
[the remark this answers is not in the document]
Comment (manitgupta) [reply expressions_api/expression_verify.go:17]: Details are here - https://docs.google.com/document/d/1bUA4fLXMRkEdHjyRblm7oj8xSrWZxwOr_6G5j1cHPJ8/edit?resourcekey=0-0YduKwYop95nupmJ8_25Zg&tab=t.0#heading=h.7owbh66tss7c

File: expressions_api/expression_verify.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}
+	err = json.Unmarshal(convJSON, convCopy)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling conv: %v", err)
+	}
+	// Reset the SpSequences field in the Conv copy.  A nil check
+	// here won't hurt but shouldn't technically be necessary as the
+	// map should always be initialized.
+	if convCopy.SpSequences != nil {
+		convCopy.SpSequences = make(map[string]ddl.Sequence) // Reset to empty map
```
Comment (manitgupta) [inline expressions_api/expression_verify.go:127]: Discussed with Astha; we will remove this logic from this PR from now and implement it together with default and check constraint expression exclusion.

File: expressions_api/expression_verify.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	spannerClient    spannerclient.SpannerClient
+	expressionDetail internal.ExpressionDetail
+}
+
+func (ev *ExpressionVerificationAccessorImpl) VerifyExpressions(ctx context.Context, verifyExpressionsInput internal.VerifyExpressionsInput) internal.VerifyExpressionsOutput {
+	err := ev.validateRequest(verifyExpressionsInput)
+	if err != nil {
+		return internal.VerifyExpressionsOutput{Err: err}
+	}
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", verifyExpressionsInput.Project, verifyExpressionsInput.Instance, ev.SpannerAccessor.SpannerClient.DatabaseName())
```
Comment (manitgupta) [inline expressions_api/expression_verify.go:39]: DatabaseName() already returns the full URI.

File: expressions_api/expression_verify.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+const THREAD_POOL = 500
+
+type ExpressionVerificationAccessor interface {
+	//Batch API which parallelizes expression verification calls
+	VerifyExpressions(ctx context.Context, verifyExpressionsInput internal.VerifyExpressionsInput) internal.VerifyExpressionsOutput
+}
+
+type ExpressionVerificationAccessorImpl struct {
+	SpannerAccessor spanneraccessor.SpannerAccessorImpl
+}
```
Comment (manitgupta) [inline expressions_api/expression_verify.go:26]: As in #936, we will implement construction of this struct in this file itself

File: expressions_api/expression_verify.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+	return verifyExpressionsOutput
+}
+
+func (ev *ExpressionVerificationAccessorImpl) verifyExpressionInternal(expressionVerificationInput ExpressionVerificationInput, mutex *sync.Mutex) common.TaskResult[internal.ExpressionVerificationOutput] {
+	var sqlStatement string
+	switch expressionVerificationInput.expressionDetail.Type {
+	case "CHECK":
+		sqlStatement = fmt.Sprintf("SELECT 1 from %s where %s;", expressionVerificationInput.expressionDetail.ReferenceElement.Name, expressionVerificationInput.expressionDetail.Expression)
+	case "DEFAULT":
+		sqlStatement = fmt.Sprintf("SELECT CAST(%s) as %s", expressionVerificationInput.expressionDetail.Expression, expressionVerificationInput.expressionDetail.ReferenceElement.Name)
```
Comment (manitgupta) [inline expressions_api/expression_verify.go:90]: `SELECT CAST(%s as %s)`

File: expressions_api/expression_verify.go (hunk 3 above)
Comment (manitgupta) [reply expressions_api/expression_verify.go:127]: Done

File: expressions_api/expression_verify.go (hunk 4 above)
Comment (manitgupta) [reply expressions_api/expression_verify.go:39]: Done

File: expressions_api/expression_verify.go (hunk 5 above)
Comment (manitgupta) [reply expressions_api/expression_verify.go:26]: Added

File: expressions_api/expression_verify.go (hunk 6 above)
Comment (manitgupta) [reply expressions_api/expression_verify.go:90]: Done

File: expressions_api/expression_verify.go (hunk 1 above)
Comment (manitgupta) [reply expressions_api/expression_verify.go:113]: Done

File: expressions_api/expression_verify.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+		verifyExpressionsOutput.Err = fmt.Errorf("%d expressions either failed verification or did not get verified. Please look at the individual errors returned for each expression", errorCount)
+
+	}
+	return verifyExpressionsOutput
+}
+
+func (ev *ExpressionVerificationAccessorImpl) verifyExpressionInternal(expressionVerificationInput ExpressionVerificationInput, mutex *sync.Mutex) task.TaskResult[internal.ExpressionVerificationOutput] {
+	var sqlStatement string
+	switch expressionVerificationInput.expressionDetail.Type {
+	case "CHECK":
```
Comment (asthamohta) [inline expressions_api/expression_verify.go:97]: I would suggest having these added to the constant file and referring it back from there to avoid errors

Comment (manitgupta) [reply expressions_api/expression_verify.go]: Done

**Review by asthamohta — APPROVED** [review]

**Pull request merged**